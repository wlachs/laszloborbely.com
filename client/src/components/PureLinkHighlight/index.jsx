import React from 'react';
import { NavLink } from 'react-router-dom';
import PureLink from '../PureLink';

function PureLinkHighlight(props) {
  const { to, children } = props;
  return (
    <NavLink to={to} className="link" activeClassName="link_active">
      {children}
    </NavLink>
  );
}

PureLinkHighlight.propTypes = PureLink.propTypes;

export default PureLinkHighlight;
