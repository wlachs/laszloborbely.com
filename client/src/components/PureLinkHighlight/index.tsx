import {NavLink} from 'react-router-dom';
import {ReactNode} from "react";

function PureLinkHighlight(props: {
    to: string;
    children?: ReactNode;
}) {
    const {to, children} = props;
    return (
        <NavLink to={to} className={({isActive}) => isActive ? "link link_active" : "link"}>
            {children}
        </NavLink>
    );
}

export default PureLinkHighlight;
