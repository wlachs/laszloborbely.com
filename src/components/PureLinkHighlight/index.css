@import '../global.css';

.link, a {
	color: var(--secondary-color);
	font-weight: 300;
	text-decoration: none !important;
	transition: 0.2s;
}

.link:hover {
	color: var(--primary-color);
}

.link_active {
	color: var(--primary-color);
}