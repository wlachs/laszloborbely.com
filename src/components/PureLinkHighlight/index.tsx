import { type ReactNode } from 'react';
import { NavLink } from 'react-router-dom';

function PureLinkHighlight(props: { to: string; children?: ReactNode }) {
	const { to, children } = props;
	return (
		<NavLink
			to={to}
			className={({ isActive }) =>
				isActive ? 'link link_active' : 'link'
			}>
			{children}
		</NavLink>
	);
}

export default PureLinkHighlight;
