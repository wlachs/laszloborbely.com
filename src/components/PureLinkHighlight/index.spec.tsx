import {cleanup, render} from '@testing-library/react';
import {MemoryRouter} from 'react-router-dom';
import {afterEach, describe, expect, it} from 'vitest';
import PureLinkHighlight from './index.tsx';

describe('PureLinkHighlight component', () => {
	afterEach(() => {
		cleanup();
	});

	it('should match snapshot', () => {
		const child = 'child';
		const {container} = render(
			<MemoryRouter>
				<PureLinkHighlight to='/'>{child}</PureLinkHighlight>,
			</MemoryRouter>,
		);
		expect(container).toMatchSnapshot();
	});

	it('should highlight link if on matching route', () => {
		const child = 'child';
		const {container} = render(
			<MemoryRouter>
				<PureLinkHighlight to='/'>{child}</PureLinkHighlight>,
			</MemoryRouter>,
		);
		expect(container.querySelector('a')?.className).toContain('link_active');
	});

	it('should not highlight link on non-matching routes', () => {
		const child = 'child';
		const {container} = render(
			<MemoryRouter>
				<PureLinkHighlight to='/blog'>{child}</PureLinkHighlight>,
			</MemoryRouter>,
		);
		expect(container.querySelector('a')?.className).not.toContain('link_active');
	});
});
