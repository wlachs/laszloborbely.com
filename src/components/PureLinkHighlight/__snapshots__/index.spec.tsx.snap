// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PureLinkHighlight component > should match snapshot 1`] = `
<div>
  <a
    aria-current="page"
    class="link link_active"
    href="/"
  >
    child
  </a>
  ,
</div>
`;
